src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2499,7 +2557,9 @@ void InterpreterSelectQuery::executeWithFill(QueryPlan & query_plan)
         if (fill_descr.empty())
             return;
 
-        auto filling_step = std::make_unique<FillingStep>(query_plan.getCurrentDataStream(), std::move(fill_descr));
+        InterpolateDescriptionPtr interpolate_descr =
+            getInterpolateDescription(query, source_header, result_header, syntax_analyzer_result->aliases, context);
```

Comment:
Why do we use a `source_header` here? I think it should be `query_plan.getCurrentDataStream().header` instead.

```
SELECT
    n,
    any(source),
    sum(inter) AS inter_s
FROM
(
    SELECT
        toFloat32(number % 10) AS n,
        'original' AS source,
        number AS inter
    FROM numbers(10)
    WHERE (number % 3) = 1
)
GROUP BY n
ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5
INTERPOLATE ( inter_s AS inter_s + 1 )

Query id: c4a37d26-ec08-42dc-9e0b-eee9a4419eb3


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 22.4.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'inter_s' while processing query: 'SELECT n, any(source), sum(inter) AS inter_s FROM (SELECT toFloat32(number % 10) AS n, 'original' AS source, number AS inter FROM numbers(10) WHERE (number % 3) = 1) GROUP BY n ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE ( inter_s AS inter_s + 1 )', required columns: 'n' 'inter_s' 'source' 'inter' 'n' 'inter_s' 'source' 'inter'. (UNKNOWN_IDENTIFIER)

```

Reply:
btw, why spaces inside interpolate expression? Let's fix formatting
`INTERPOLATE ( inter_s AS inter_s + 1 )` -> `INTERPOLATE (inter_s AS inter_s + 1)`

Reply:
actually I need columns which participate in interpolate expressions - they can be from source_header and result_header, but getCurrentDataStream().header seems the same as result_header